D/DuckDB/build_tarballs.jl

```diff
@@ -10,8 +10,7 @@ sources = [
     GitSource("https://github.com/duckdb/duckdb.git", "f680b7d08f56183391b581077d4baf589e1cc8bd"),
 ]
 
-# Bash recipe for building across all platforms
-script = raw"""
+script_nonwindows = raw"""
```

Comment:
Why not using conditionals inside the script? Having different scripts for different platforms is a needless complication.

Reply:
I figured that the cases inside the script would have be decided based on the target, which seemed to be defined to this julia array:
```julia
julia> using BinaryBuilder; expand_cxxstring_abis(supported_platforms())
┌ Warning: Running Registrator on a Julia version that does not support weak dependencies. Weak dependencies will not be registered.
└ @ RegistryTools C:\Users\dennis.bal\.julia\packages\RegistryTools\tHkoa\src\RegistryTools.jl:18
29-element Vector{Platform}:
 Linux i686 {cxxstring_abi=cxx03, libc=glibc}
 Linux i686 {cxxstring_abi=cxx11, libc=glibc}
 Linux x86_64 {cxxstring_abi=cxx03, libc=glibc}
 Linux x86_64 {cxxstring_abi=cxx11, libc=glibc}
 Linux aarch64 {cxxstring_abi=cxx03, libc=glibc}
 Linux aarch64 {cxxstring_abi=cxx11, libc=glibc}
 Linux armv6l {call_abi=eabihf, cxxstring_abi=cxx03, libc=glibc}
 Linux armv6l {call_abi=eabihf, cxxstring_abi=cxx11, libc=glibc}
 Linux armv7l {call_abi=eabihf, cxxstring_abi=cxx03, libc=glibc}
 Linux armv7l {call_abi=eabihf, cxxstring_abi=cxx11, libc=glibc}
 Linux powerpc64le {cxxstring_abi=cxx03, libc=glibc}
 Linux powerpc64le {cxxstring_abi=cxx11, libc=glibc}
 Linux i686 {cxxstring_abi=cxx03, libc=musl}
 Linux i686 {cxxstring_abi=cxx11, libc=musl}
 Linux x86_64 {cxxstring_abi=cxx03, libc=musl}
 Linux x86_64 {cxxstring_abi=cxx11, libc=musl}
 Linux aarch64 {cxxstring_abi=cxx03, libc=musl}
 Linux aarch64 {cxxstring_abi=cxx11, libc=musl}
 Linux armv6l {call_abi=eabihf, cxxstring_abi=cxx03, libc=musl}
 Linux armv6l {call_abi=eabihf, cxxstring_abi=cxx11, libc=musl}
 Linux armv7l {call_abi=eabihf, cxxstring_abi=cxx03, libc=musl}
 Linux armv7l {call_abi=eabihf, cxxstring_abi=cxx11, libc=musl}
 macOS x86_64
 macOS aarch64
 FreeBSD x86_64
 Windows i686 {cxxstring_abi=cxx03}
 Windows i686 {cxxstring_abi=cxx11}
 Windows x86_64 {cxxstring_abi=cxx03}
 Windows x86_64 {cxxstring_abi=cxx11}
 ```
 
 Where there are e.g. multiple names for windows that have to be concidered. Furthermore, it is unclear how those `Platform` objects are passed to bash.
 
 I figured it was actually extremely neat and tidy to move that logic into julia, using clearly named and easy-to-use function is Sys.iswindows, instead of adding string matching for multiple cases, which to me felt hacky. So we seem to have different intuitions from looking at the same stuff.

That said, you are the desciding vote, and I am happy to implement it whatever way you deem best. I am however possibly not capable of doing it the way you want, so you might have to make the change yourself.

Reply:
@KronosTheLate The platform being built is available in `$target` (and in `$bb_full_target`).

Reply:
This is also explained in the docs:
* https://docs.binarybuilder.org/stable/build_tips/#Initiating-different-shell-commands-based-on-target
* https://docs.binarybuilder.org/stable/building/#Platforms

Reply:
> In any case, both options are far less understandable to me than what I have done.

What you've done is wrong because you'll register the package twice. Also, what you've done has duplicate code, which is error prone and less maintainable.  Please use the conditionals inside the script instead.